File: src/cpp/http/purge.cpp
#include "purge.h"
#include "sdk-sys.h"

namespace fastly::http::purge {

fastly::expected<void> purge_surrogate_key(std::string_view surrogate_key) {
  fastly::sys::error::FastlyError *err;
  fastly::sys::http::purge::f_http_purge_purge_surrogate_key(
      static_cast<std::string>(surrogate_key), err);
  if (err != nullptr) {
    return fastly::unexpected(err);
  } else {
    return fastly::expected<void>();
  }
}

fastly::expected<void>
soft_purge_surrogate_key(std::string_view surrogate_key) {
  fastly::sys::error::FastlyError *err;
  fastly::sys::http::purge::f_http_purge_soft_purge_surrogate_key(
      static_cast<std::string>(surrogate_key), err);
  if (err != nullptr) {
    return fastly::unexpected(err);
  } else {
    return fastly::expected<void>();
  }
}

} // namespace fastly::http::purge

File: src/http/purge.rs
use cxx::CxxString;

use crate::{error::ErrPtr, try_fe};

pub fn f_http_purge_purge_surrogate_key(surrogate_key: &CxxString, mut err: ErrPtr) {
    try_fe!(
        err,
        fastly::http::purge::purge_surrogate_key(try_fe!(err, surrogate_key.to_str()))
    );
}

pub fn f_http_purge_soft_purge_surrogate_key(surrogate_key: &CxxString, mut err: ErrPtr) {
    try_fe!(
        err,
        fastly::http::purge::soft_purge_surrogate_key(try_fe!(err, surrogate_key.to_str()))
    );
}
